packages/client/tests/functional/_utils/setupTestSuiteMatrix.ts

```diff
@@ -35,11 +43,38 @@ export type TestSuiteMeta = ReturnType<typeof getTestSuiteMeta>
  *
  * @param tests where you write your tests
  */
-function setupTestSuiteMatrix(tests: (suiteConfig: TestSuiteConfig, suiteMeta: TestSuiteMeta) => void) {
+function setupTestSuiteMatrix(
+  tests: (suiteConfig: TestSuiteConfig, suiteMeta: TestSuiteMeta) => void,
+  options?: MatrixOptions,
+) {
   const originalEnv = process.env
   const suiteMeta = getTestSuiteMeta()
-  const suiteTable = getTestSuiteTable(suiteMeta)
+  const suiteConfig = getTestSuiteConfigs(suiteMeta)
+  const suiteTable = getTestSuiteTable(suiteMeta, suiteConfig)
   const forceInlineSnapshot = process.argv.includes('-u')
+  const suiteConfigProviders = suiteConfig.map(({ provider }) => provider)
+
+  const missingProviders = Object.values(Providers)
```

Comment:
I'd like to extract this into a utility function instead (+use reduce instead of map+filter)

```ts
/**
 * Ensure that we are not forgetting to add a provider to the matrix unless we
 * explicitly opt out during in the {@link setupTestSuiteMatrix} creation.
 * @param suiteConfigs
 * @param suiteMeta
 * @param options
 */
function checkMissingProviders(suiteConfigs: TestSuiteConfig[], suiteMeta: TestSuiteMeta, options: MatrixOptions = {}) {
  const suiteConfigProviders = suiteConfigs.map(({ provider }) => provider)

  const missingProviders = Object.values(Providers).reduce((acc, provider) => {
    if (suiteConfigProviders.includes(provider)) return acc
    if (options.optOut?.from.includes(provider)) return acc

    return [...acc, provider]
  }, [] as Providers[])

  if (missingProviders.length) {
    throw new Error(
      `Test: '${suiteMeta.testDirName}' is missing providers '${missingProviders
        .map((x) => `'${x}'`)
        .join(', ')}' out out using options.optOut`,
    )
  }
}
```

Reply:
Nice idea, added https://github.com/prisma/prisma/pull/13507/files#diff-96bd19e93b8f591ba66f10ae527c63fd90a115bde817d872d378cfce788a210f